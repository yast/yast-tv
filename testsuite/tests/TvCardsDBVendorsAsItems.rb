# encoding: utf-8

module Yast
  class TvCardsDBVendorsAsItemsClient < Client
    def main
      # testedfiles: Tv.ycp

      Yast.include self, "testsuite.rb"

      @READ = { "target" => { "size" => -1, "tmpdir" => "/tmp" } }

      TESTSUITE_INIT([@READ], nil)
      Yast.import "Tv"

      TEST_It_Now(nil, nil, nil)

      # preselected should be the vendor of autodetected card
      TEST_It_Now(
        {
          "module"     => ["bttv"],
          "name"       => "My Tv_card",
          "unique_key" => "123xyz"
        },
        nil,
        nil
      )

      # preselected should be the vendor of card 64
      TEST_It_Now(
        { "name" => "my autodetected card" },
        { "card" => "64" },
        "bttv"
      )

      # preselected card has also number 64, but another module (and vendor)
      TEST_It_Now(
        { "name" => "my autodetected card" },
        { "card" => "64" },
        "other"
      )

      nil
    end

    def TEST_It_Now(auto, params, mod)
      auto = deep_copy(auto)
      params = deep_copy(params)
      DUMP("---------------")

      Tv.cards_database = []
      items = Tv.CardsDBVendorsAsItems(auto, params, mod)
      DUMP(items)

      Tv.cards_database = [
        {
          "cards" => [
            {
              "module"     => ["bttv"],
              "name"       => "Unknown bttv card",
              "parameters" => { "card" => "0" }
            },
            {
              "module"     => ["bttv"],
              "name"       => "TView99 CPH06X",
              "parameters" => { "card" => "38" }
            },
            {
              "module"     => ["other"],
              "name"       => "Some card with id 64",
              "parameters" => { "card" => "64" }
            }
          ],
          "name"  => "Other vendors"
        },
        {
          "cards" => [
            {
              "module"     => ["bttv"],
              "name"       => "ATI TV-Wonder VE",
              "parameters" => { "card" => "64" }
            }
          ],
          "name"  => "ATI"
        }
      ]
      items = Tv.CardsDBVendorsAsItems(auto, params, mod)
      DUMP(items)

      nil
    end
  end
end

Yast::TvCardsDBVendorsAsItemsClient.new.main
