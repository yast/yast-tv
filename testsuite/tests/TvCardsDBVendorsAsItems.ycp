{

    include "testsuite.ycp";

    map READ = $[
	"target" : $[ "size": -1, "tmpdir" : "/tmp", ]
    ];

    TESTSUITE_INIT([READ], nil);
    import "Tv";

    define void TEST_It_Now(map auto, map params, string mod) ``{

	DUMP("---------------");

	Tv::cards_database = [];
	list items = Tv::CardsDBVendorsAsItems (auto,params, mod);
	DUMP(items);

	Tv::cards_database = [
	    $["cards": [
		$["module":"bttv", "name":"Unknown bttv card",
		  "parameters":$["card":"0"]],
		$["module":"bttv", "name":"TView99 CPH06X",
		  "parameters":$["card":"38"]],
		$["module":"other", "name":"Some card with id 64",
		  "parameters":$["card":"64"]]
	      ],
	      "name":"Other vendors"
	    ],
	    $["cards": [
		$["module":"bttv", "name":"ATI TV-Wonder VE",
		  "parameters":$["card":"64"]]
	      ],
	      "name":"ATI"
	    ]
	];
	items = Tv::CardsDBVendorsAsItems (auto, params, mod);
	DUMP(items);

    }

    TEST_It_Now(nil, nil,nil);

    // preselected should be the vendor of autodetected card
    TEST_It_Now( $["module":"bttv",
				   "name":"My Tv_card",
				   "unique_key":"123xyz"
				 ],nil,nil);

    // preselected should be the vendor of card 64
    TEST_It_Now( $["name":"my autodetected card" ], $["card":"64"], "bttv");

    // preselected card has also number 64, but another module (and vendor)
    TEST_It_Now( $["name":"my autodetected card" ], $["card":"64"], "other");

}
