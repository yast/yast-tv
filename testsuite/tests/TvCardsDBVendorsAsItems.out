Read	.target.tmpdir "/tmp"
Read	.target.size "/etc/install.inf" -1
Read	.probe.architecture nil
Read	.probe.has_apm nil
Read	.probe.has_pcmcia nil
Read	.probe.has_smp nil
Read	.probe.system nil
Dump	---------------
Dump	[[], nil]
Dump	[[`item (`id (0), "Other vendors", false), `item (`id (1), "ATI", false)], nil]
Dump	---------------
Dump	[[`item (`id (0), "Autodetected card", true)], 0]
Dump	[[`item (`id (0), "Autodetected card", true), `item (`id (1), "Other vendors", false), `item (`id (2), "ATI", false)], 0]
Dump	---------------
Dump	[[`item (`id (0), "Autodetected card", false)], nil]
Dump	[[`item (`id (0), "Autodetected card", false), `item (`id (1), "Other vendors", false), `item (`id (2), "ATI", true)], 2]
Dump	---------------
Dump	[[`item (`id (0), "Autodetected card", false)], nil]
Dump	[[`item (`id (0), "Autodetected card", false), `item (`id (1), "Other vendors", true), `item (`id (2), "ATI", false)], 1]
